distribution/openrct2.d.ts

```diff
@@ -265,10 +265,10 @@ declare global {
          * @param execute Logic for validating and executing the action.
          * @throws An error if the action has already been registered by this or another plugin.
          */
-        registerAction(
+        registerAction<T = object>(
             action: string,
-            query: (args: object) => GameActionResult,
-            execute: (args: object) => GameActionResult): void;
+            query: (args: GameActionEventArgs<T>) => GameActionResult,
```

Comment:
Why `GameActionEventArgs` and not just `GameActionArgs`? What is so "event" about these args?

Reply:
Ah, I see. This interface is already present.

Reply:
This interface was chosen to match subscribe callbacks. Unfortunately I didn't choose the naming.